ambry-tools/src/main/java/com.github.ambry/account/AccountTool.java

```diff
@@ -0,0 +1,542 @@
+ * </p>
+ * <p>
+ *   A sample usage of the tool is:
+ *   <code>
+ *     java -Dlog4j.configuration=file:./log4j.properties -cp ambry.jar com.github.ambry.account.AccountTool
+ *     --zkServer localhost:2181 --storePath /ambry/test/helixPropertyStore --zkLayoutPath ./zklayout.json --clustername cluster1 --dcname dc1
+ *   </code>
+ * </p>
+ * <p>
+ *   The file for dc zkinfo looks like this
+ *   <pre>
+ *     {
+ *     "zkInfo": [
+ *         {
+ *             "datacenter": "ei4",
+ *             "id": 3,
+ *             "zkConnectStr": "zk-ei4-shared.int.linkedin.com:12913"
+ *         }
+ *     ]
+ *    }
+ *   </pre>
+ *   The dcname parameter should be one of the "datacenter" in the list.
+ * </p>
+ *
+ * <p>
+ *   The file for the accounts to create/update contains a {@link org.json.JSONArray} of account in its json form. Invalid
+ *   records will fail the operation. For example the file is in the following format:
+ *   <pre>
+ *     [
+ *      {
+ *        "accountId": 1,
+ *        "accountName": "account1",
+ *        "containers": [
+ *          {
+ *            "parentAccountId": 1,
+ *            "containerName": "container1",
+ *            "description": "This is the first container of account1",
+ *            "isPrivate": false,
+ *            "containerId": 1,
+ *            "version": 1,
+ *            "status": "ACTIVE"
+ *          }
+ *        ],
+ *        "version": 1,
+ *        "status": "ACTIVE"
+ *      },
+ *      {
+ *        "accountId": 2,
+ *        "accountName": "account1",
+ *        "containers": [
+ *          {
+ *            "parentAccountId": 2,
+ *            "containerName": "container1",
+ *            "description": "This is the first container of ",
+ *            "isPrivate": true,
+ *            "containerId": 1,
+ *            "version": 1,
+ *            "status": "INACTIVE"
+ *          }
+ *        ],
+ *        "version": 1,
+ *        "status": "ACTIVE"
+ *      }
+ *     ]
+ *   </pre>
+ * </p>
+ */
+public class AccountTool {
+  private final MetricRegistry registry;
+  private final BackupFileManager backupFileManager;
+  private final VerifiableProperties verifiableProperties;
+  private final HelixPropertyStoreConfig storeConfig;
+  private final HelixAccountServiceConfig accountServiceConfig;
+  private final HelixPropertyStore<ZNRecord> helixStore;
+
+  private RouterStore routerStore = null;
+  private Router router = null;
+  private ClusterMap clusterMap = null;
+  private HelixAccountService accountService = null;
+
+  private static final int ZK_CLIENT_CONNECTION_TIMEOUT_MS = 5000;
+  private static final int ZK_CLIENT_SESSION_TIMEOUT_MS = 20000;
+  private static final String DEFAULT_HOSTNAME = "localhost";
+
+  private String clusterName = null;
+  private String hostname = null;
+  private String dcName = null;
+  private String clusterMapDcsZkConnectString = null;
+
+  private static final List<String> ACTION_CHOICES = Arrays.asList("list", "view", "rollback", "update");
```

Comment:
Can we change this to Enum? Refer to `DumpIndexTool.java` and i think enum `Operation` there is an example.